### PR TITLE
Email Field Improvements

### PR DESCRIPTION
**Pull request by Andrew-Dickinson**
This PR is a chonker, but is best viewed commit-by-commit. It's split into three different changes, with related motivations:

1. Refactor `email_address` -> `primary_email_address` & `secondary_emails` -> `additional_email_addresses`
  a. Previously we had `email_address`, which is ambiguous for downstream consumers of our data, particularly when presented alongside the upcoming `all_email_addresses` field. This renaming makes it clear that this email is the primary email, but not necessarily the only one, and makes the motivation for the  `all_email_addresses` field more clear
2. Add the aforementioned `all_email_addresses` field to the `Member` object, and in the `MemberSerializer`
    a. This field captures all email addresses and presents them to API consumers in a single array, closing #117 
3. Do email lookups based on all email fields
   a. When we need to lookup a member based on email address, especially for de-duplication, we really shouldn't look just at the primary email field. This change modifies our filtering logic to search on all email fields
   b. Also adds tests for the `/lookup` endpoints, since these were missing. This is progress towards #148

This will need to be rebased onto #189, since we add a migration

### CODE REVIEW

File: src/meshapi/models.py [hunk 1]
```diff
@@ -60,6 +62,22 @@ def __str__(self):
             return self.name
         return f"MeshDB Member ID {self.id}"
 
+    @property
+    def all_email_addresses(self) -> List[str]:
+        all_emails = []
+        if self.primary_email_address and self.primary_email_address not in all_emails:
+            all_emails.append(self.primary_email_address)
```
[the remark this answers is not in the document]
Comment (WillNilges) [reply src/meshapi/models.py:69]: ChatGPT suggested this. Though Sets are generally less performant than Lists, you're basically doing the deduplication manually rn.

```python
@property
def all_email_addresses(self) -> List[str]:
    all_emails = {self.primary_email_address, self.stripe_email_address}
    all_emails.update(self.additional_email_addresses)
    return list(filter(None, all_emails))
```

File: src/meshapi/tests/test_join_form.py [hunk 2]
```diff
@@ -226,34 +233,118 @@ def test_bad_address_join_form(self):
...
+            json.loads(
+                response1.content.decode("utf-8"),
+            )["member_id"],
+            json.loads(
+                response2.content.decode("utf-8"),
+            )["member_id"],
         )
 
-        validate_successful_join_form_submission(self, "Valid Join Form", s, response)
+    def test_member_moved_and_used_stripe_email_join_form(self):
```
Comment (WillNilges) [inline src/meshapi/tests/test_join_form.py:274]: damn nice edge case

File: src/meshapi/tests/test_lookups.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import json
+
+from django.contrib.auth.models import User
+from django.test import Client, TestCase
+
+from ..models import Member
+from .sample_data import sample_member
+
+
+class TestMemberLookups(TestCase):
```
Comment (WillNilges) [inline src/meshapi/tests/test_lookups.py:10]: i love you

**Review by WillNilges — APPROVED** [review]
ship it

File: src/meshapi/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Andrew-Dickinson) [reply src/meshapi/models.py:69]: That will break the order. I think it's nice to have the primary email always in the first spot in the array 🙃

File: src/meshapi/tests/test_join_form.py (hunk 2 above)
Comment (Andrew-Dickinson) [reply src/meshapi/tests/test_join_form.py:274]: <3

Needed to get coverage on my new feature somehow

File: src/meshapi/tests/test_lookups.py (hunk 3 above)
Comment (Andrew-Dickinson) [reply src/meshapi/tests/test_lookups.py:10]: Again, it was a bit selfish, I needed coverage for my new code 🙃
